File: data/src/main/java/com/ericafenyo/data/entities/Article.java
package com.ericafenyo.data.entities;

public class Article {
    private String title;
    private String section;
    private String description;
    private String imageUrl;
    private String articleUrl;
    private String publishedDate;
    private String author;

    public Article(String title, String section, String description, String imageUrl,
        String articleUrl, String publishedDate, String author) {
        this.title = title;
        this.section = section;
        this.description = description;
        this.imageUrl = imageUrl;
        this.articleUrl = articleUrl;
        this.publishedDate = publishedDate;
        this.author = author;
    }

    public String getTitle() {
        return title;
    }

    public String getSection() {
        return section;
    }

    public String getDescription() {
        return description;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public String getArticleUrl() {
        return articleUrl;
    }

    public String getPublishedDate() {
        return publishedDate;
    }

    public String getAuthor() {
        return author;
    }

    public static final class Builder {

        private String title;
        private String section;
        private String description;
        private String imageUrl;
        private String articleUrl;
        private String publishedDate;
        private String author;

        public Builder() {}

        public Builder title(String title) {
            this.title = title;
            return this;
        }

        public Builder section(String section) {
            this.section = section;
            return this;
        }

        public Builder description(String description) {
            this.description = description;
            return this;
        }

        public Builder imageUrl(String imageUrl) {
            this.imageUrl = imageUrl;
            return this;
        }

        public Builder articleUrl(String articleUrl) {
            this.articleUrl = articleUrl;
            return this;
        }

        public Builder publishedDate(String publishedDate) {
            this.publishedDate = publishedDate;
            return this;
        }

        public Builder author(String author) {
            this.author = author;
            return this;
        }

        public Article build() {
            return new Article(
                title,
                section,
                description,
                imageUrl,
                articleUrl,
                publishedDate,
                author
            );
        }
    }
}

File: data/src/main/java/com/ericafenyo/data/entities/Article.kt
/*
 * Copyright (C) 2019 Eric Afenyo
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.ericafenyo.data.article.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * A Room Entity used to create a Database table.
 * */
@Entity(tableName = "article")
data class Article(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val title: String,
    val section: String,
    val description: String,
    @ColumnInfo(name = "image_url") val imageUrl: String,
    @ColumnInfo(name = "article_url") val articleUrl: String,
    @ColumnInfo(name = "published_date") val publishedDate: String,
    val author: String
)